src/components/application_manager/test/rpc_passing_handler_test.cc

```diff
@@ -0,0 +1,447 @@
+        true;
+    record[am::strings::service_manifest][am::strings::handled_rpcs] =
+        smart_objects::SmartObject(smart_objects::SmartType::SmartType_Array);
+    record[am::strings::service_manifest][am::strings::handled_rpcs][0] =
+        mobile_apis::FunctionID::SendLocationID;
+    record[am::strings::service_published] = true;
+    record[am::strings::service_active] = true;
+    app_service.record = record;
+
+    return app_service;
+  }
+
+  smart_objects::SmartObject CreatePassThroughRequest(uint32_t connection_key,
+                                                      int32_t correlation_id) {
+    smart_objects::SmartObject record(smart_objects::SmartType::SmartType_Map);
+    record[am::strings::params][am::strings::connection_key] = connection_key;
+    record[am::strings::params][am::strings::correlation_id] = correlation_id;
+    record[am::strings::params][am::strings::function_id] =
+        mobile_apis::FunctionID::SendLocationID;
+    record[am::strings::params][am::strings::message_type] =
+        am::MessageType::kRequest;
+    record[am::strings::msg_params][am::strings::latitude_degrees] = 50;
+    record[am::strings::msg_params][am::strings::longitude_degrees] = 50;
+    record[am::strings::msg_params][am::strings::location_name] =
+        "TestLocation";
+
+    return record;
+  }
+
+  smart_objects::SmartObject CreatePassThroughResponse(
+      uint32_t connection_key,
+      int32_t correlation_id,
+      const std::string& result_code,
+      bool success,
+      std::string info = std::string()) {
+    smart_objects::SmartObject record(smart_objects::SmartType::SmartType_Map);
+    record[am::strings::params][am::strings::connection_key] = connection_key;
+    record[am::strings::params][am::strings::correlation_id] = correlation_id;
+    record[am::strings::params][am::strings::function_id] =
+        mobile_apis::FunctionID::SendLocationID;
+    record[am::strings::params][am::strings::message_type] =
+        am::MessageType::kResponse;
+    record[am::strings::msg_params][am::strings::result_code] = result_code;
+    record[am::strings::msg_params][am::strings::success] = success;
+    record[am::strings::msg_params][am::strings::info] = info;
+
+    return record;
+  }
+
+  void SendRequestToASP(const PassthroughParams& params, uint32_t timeout = 0) {
+    if (timeout == 0) {
+      // If timeout argument is not sepcified
+      timeout = kRPCPassthroughTimeout;
+    }
+
+    {
+      InSequence dummy;
+      // Call RPCPassThrough with request smart object
+
+      // Will call PopulateRPCRequestQueue
+      EXPECT_CALL(mock_app_service_manager_, GetActiveServices())
+          .WillOnce(Return(app_services_));
+      EXPECT_CALL(mock_app_manager_,
+                  IncreaseForwardedRequestTimeout(params.connection_key,
+                                                  params.correlation_id))
+          .Times(app_services_.size());
```

Comment:
Hmmm, I suppose I didn't realize that this was how this worked before. This is incorrect, this function should be called each time that the RPC is passed to another service, not several times when preparing passthrough. I believe that this means that the timeout doesn't actually work now, because it doesn't just add to the timeout each time this is called, it resets the timeout to `default_timeout + rpc_passthrough_timeout`, meaning that this is the total time that the app has before a GENERIC_ERROR response.

Reply:
Actually never mind, this is called before this message is added to the `RequestController`, so it won't time out because of this. That said, this call will have absolutely no effect, so it should be removed.

Reply:
Since this fix will require changes to the `rpc_passing_handler.cc` and the test file it should probably be in its own PR.

Reply:
Should still remove this expectation, since it doesn't have a return value anyway.

Reply:
Resolved in https://github.com/smartdevicelink/sdl_core/pull/2912/commits/7d556a33dc13e3411e22c3e07eaad330c45a1e8b